client/go/outline/config/config_iptable_test.go

```diff
@@ -0,0 +1,200 @@
+// Copyright 2025 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/Jigsaw-Code/outline-apps/client/go/configyaml"
+	"github.com/Jigsaw-Code/outline-sdk/transport"
+	"github.com/stretchr/testify/require"
+)
+
+func TestParseIPTableStreamDialer(t *testing.T) {
+	ctx := context.Background()
+
+	// Define a set of mock dialers for our tests to use.
+	parseSE := func(ctx context.Context, config configyaml.ConfigNode) (*Dialer[transport.StreamConn], error) {
+		configMap, ok := config.(map[string]any)
+		if !ok {
+			return nil, errors.New("config is not a map[string]any")
+		}
+
+		name, ok := configMap["name"].(string)
+		if !ok {
+			return nil, errors.New("mock dialer config must have a 'name'")
+		}
+
+		var dialer transport.StreamDialer
+		connType := ConnTypeTunneled
+		switch name {
+		case "dialerA":
+			dialer = &errorStreamDialer{name: "dialerA"}
+		case "dialerB":
+			dialer = &errorStreamDialer{name: "dialerB"}
+		case "default":
+			dialer = &errorStreamDialer{name: "default"}
+		case "direct":
+			dialer = &errorStreamDialer{name: "direct"}
+			connType = ConnTypeDirect
+		default:
+			return nil, fmt.Errorf("no mock dialer found with name: %s", name)
+		}
+		return &Dialer[transport.StreamConn]{Dial: dialer.DialStream, ConnectionProviderInfo: ConnectionProviderInfo{ConnType: connType}}, nil
+	}
+
+	// Define test cases
+	testCases := []struct {
+		name             string
+		configYAML       string
+		expectErr        string
+		checkDialer      func(*testing.T, *Dialer[transport.StreamConn])
+		expectedConnType ConnType
+	}{
+		{
+			name: "Happy Path - valid config",
+			configYAML: `
+table:
+  - ip: 192.168.1.0/24
+    dialer: {name: dialerA}
+  - ip: 10.0.0.1
+    dialer: {name: dialerB}
+  - ip: 0.0.0.0/0
```

Comment:
Can you add an example without the `ip`?

I would like the user to not have to specify the IP for a catch all case. Please make sure that works.

Reply:
Having to write `0.0.0.0/0` is error prone and people won't really know. It's also confusing, since it's IPv4, but we need IPv6 too. We need a way to represent a "catch all".


It could be an entry with an empty IP, or a fallback dialer option.

Empty IP:

```yaml
table:
  - ip: 192.168.1.0/24
    dialer: {name: dialerA}
  - ip: 10.0.0.1
    dialer: {name: dialerB}
  - dialer: {name: default}
```

Fallback dialer

```yaml
table:
  - ip: 192.168.1.0/24
    dialer: {name: dialerA}
  - ip: 10.0.0.1
    dialer: {name: dialerB}

fallback:
  name: default
```

@jyyi1 thoughts?